### PR TITLE
ICU-22692 Update names of SimpleNumber functions

### PR DESCRIPTION
**Pull request by sffc**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22692
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [ ] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Comment by markusicu** [comment]
@sffc please remember to finish this asap before code freeze

**Comment by sffc** [comment]
it's high on my stack

**Comment by markusicu** [comment]
This affects API. It would be really good to get done in 75, but for that it is turning into a blocker for the rc branch and BRS work. If it slips to 76, then that delays when all of SimpleNumber becomes stable.

@richgillam FYI

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 1]
```diff
@@ -68,44 +68,49 @@ class U_I18N_API SimpleNumber : public UMemory {
...
      *
-     * This function immediately mutates the inner value.
+     * @draft ICU 75
+     */
+    void setMaximumIntegerDigits(uint32_t maximumIntegerDigits, UErrorCode& status);
+#endif // U_HIDE_DRAFT_API
+
+#ifndef U_HIDE_DEPRECATED_API
+    /** Alias for setMaximumIntegerDigits.
      *
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:92]:
```suggestion
    /**
     * Alias for setMaximumIntegerDigits.
     * Will be removed after ICU 75.
     *
```

File: icu4c/source/i18n/unicode/usimplenumberformatter.h [hunk 2]
```diff
@@ -152,25 +150,32 @@ usnum_setMinimumIntegerDigits(USimpleNumber* unumber, int32_t minimumIntegerDigi
...
+ * @draft ICU 75
+ */
+U_CAPI void U_EXPORT2
+usnum_setMaximumIntegerDigits(USimpleNumber* unumber, int32_t maximumIntegerDigits, UErrorCode* ec);
+#endif // U_HIDE_DRAFT_API
+
+
+#ifndef U_HIDE_DEPRECATED_API
+/** Alias for setMaximumIntegerDigits.
  *
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/usimplenumberformatter.h:172]:
```suggestion
/**
 * Alias for setMaximumIntegerDigits.
 * Will be removed after ICU 75.
 *
```

File: icu4c/source/i18n/unicode/simplenumberformatter.h [hunk 3]
```diff
@@ -68,44 +68,49 @@ class U_I18N_API SimpleNumber : public UMemory {
...
-     * @draft ICU 73
+     * @deprecated ICU 75
      */
-    void truncateStart(uint32_t maximumIntegerDigits, UErrorCode& status);
+    void truncateStart(uint32_t maximumIntegerDigits, UErrorCode& status) {
+        setMaximumIntegerDigits(maximumIntegerDigits, status);
```
Comment (markusicu) [inline icu4c/source/i18n/unicode/simplenumberformatter.h:96]: This does not compile if U_HIDE_DRAFT_API but not U_HIDE_DEPRECATED_API.
Please try to move line 88 `#endif // U_HIDE_DRAFT_API` to after line 98 `#endif // U_HIDE_DEPRECATED_API`

**Review by richgillam — APPROVED** [review]
LGTM

**Comment by markusicu** [comment]
@sffc please make the requested changes which should also fix the CI checks

**Review by richgillam — APPROVED** [review]
Still LGTM.

**Comment by markusicu** [comment]
> Oops: Nesting the two header guards results in a different failure
> I guess I should just implement it in C++ and not try to clevery implement the wrapping in the .h?

Sorry... I guess we do need the hide-draft and hide-deprecated sections to not overlap.
You can move the truncateStart() definition into the .cpp file, or maybe you can *declare* it inline, and define it later in the .h file guarded by `#if !defined(U_HIDE_DRAFT_API) && !defined(U_HIDE_DEPRECATED_API)`

**Review by markusicu — APPROVED** [review]

**Review by richgillam — APPROVED** [review]

**Pull request merged**